doc/source/usage.rst

```diff
@@ -4,17 +4,43 @@ Usage
 API overview
 ++++++++++++
 
+A Filehandle (:c:type:`DcmFilehandle`) enables access of a `DICOM file
+<http://dicom.nema.org/medical/dicom/current/output/chtml/part10/chapter_3.html#glossentry_DICOMFile>`_,
+which contains an encoded Data Set representing a SOP Instance.
+A Filehandle can be created via :c:func:`dcm_filehandle_create_from_file()`
+or :c:func:`dcm_filehandle_create_from_memory()` , and destroyed via
+:c:func:`dcm_filehandle_destroy()`.  You can make your own load functions
+to load from other IO sources, see :c:func:`dcm_filehandle_create()`.
+
+The content of a Part10 file can be read using various functions.
+
+The `File Meta Information
+<http://dicom.nema.org/medical/dicom/current/output/chtml/part10/chapter_3.html#glossentry_FileMetaInformation>`_
+can be accessed via :c:func:`dcm_filehandle_get_file_meta()`.
+
+The principal metadata of the Data Set can be accessed via
+:c:func:`dcm_filehandle_get_metadata()`. This function will stop read on tags
+which are likely to take a long time to process.
```

Comment:
Yes, it'll stop on any of these tags:

```C
const DcmDataSet *dcm_filehandle_get_metadata(DcmError **error,
                                              DcmFilehandle *filehandle)
{
    // we stop on any of the tags that start a huge group that
    // would take a long time to parse
    static uint32_t stop_tags[] = {
        TAG_REFERENCED_IMAGE_NAVIGATION_SEQUENCE,
        TAG_PER_FRAME_FUNCTIONAL_GROUP_SEQUENCE,
        TAG_PIXEL_DATA,
        TAG_FLOAT_PIXEL_DATA,
        TAG_DOUBLE_PIXEL_DATA,
        0,
    };
```

I wanted to avoid listing the tags explicitly so people didn't depend on this behaviour and we could change it in the future without breaking compatibility.

Reply:
The `TAG_PER_FRAME_FUNCTIONAL_GROUP_SEQUENCE` is unfortunately very bulky, but strictly shouldn't be considered a bulkdata element (the sequence is large but the individual nested elements are small).

I suggest loading this metadata element, even if it may take a bit to parse.

Reply:
It can take many, many seconds to parse, unfortunately. 

I tried a test with a 1.9gb Leica DICOM I have:

```
john@banana ~/pics/openslide/dicom/272320e7-5abb-ccac-8f8b-9c2008f53845_114626 $ time openslide-show-properties 1.3.6.1.4.1.36533.1214640142408762111756320324051133413.dcm > log

real	0m0.083s
user	0m0.012s
sys	0m0.071s
```

So about 0.1s to assemble the files into a pyramid and get the metadata. If I disable the skip-per-frame-functional-group-sequence feature I see:

```
john@banana ~/pics/openslide/dicom/272320e7-5abb-ccac-8f8b-9c2008f53845_114626 $ time openslide-show-properties 1.3.6.1.4.1.36533.1214640142408762111756320324051133413.dcm > log
real	0m5.412s
user	0m5.423s
sys	0m0.493s
```

Over 5s, so around 50x slower. I think a 5s pause on file open would make interactive viewing of DICOM files frustrating, and there's no benefit, since the information is not useful.